### ISSUE TITLE
Queries regarding the event handlers for react-menu

### ISSUE THREAD

**Issue by aditya-padhi-kbl** [szhsin/react-menu#675]
React/React-dom version: 16.11
React-Menu version: 2.3.4

CodeSandbox URL:- https://codesandbox.io/s/react-menu-styled-components-forked-f40ng5?file=/src/App.js

Hi @szhsin,
Thank-you for this great react component.  I came across a use case so thought of asking you .
When I click on an anchor element I am setting `setState('open')`, however when I click on the button again, when I check the state value it is still `closed`. Is it the expected behavior of the react-menu component ?

Similarly for `SubMenu`, I am setting the following prop:- `openTrigger="clickOnly"`, it is working as expected, however when I click on the open `SubMenu` then it is not closing.

Can you please suggest if this is the expected behavior ?

Kind Regards,
Aditya

**Comment by szhsin** [comment]
Hello @aditya-padhi-kbl , it looks like you want the button to toggle menu between open and close. 

I noticed in your example a `ControlledMenu` component is used and toggle is implemented using the `state` prop. Instead of doing this, the `Menu` component supports the toggle behaviour out of box and it's easier to use.

I have made some updates to your example and hopefully that works for you.
https://codesandbox.io/s/react-menu-styled-components-forked-1297cy?file=/src/App.js 

For submenu, when it is open and clicked again, it will stay open. This is an expected behaviour which aligns with most system menu behaviour. E.g., take a look of the submenu on macOS it behaves in this way.
